tools/confix/migrations.go

```diff
@@ -0,0 +1,21 @@
+package confix
+
+import (
+	"github.com/creachadair/tomledit/transform"
+)
+
+const (
+	AppConfig    = "app.toml"
+	ClientConfig = "client.toml"
+	TMConfig     = "config.toml"
+)
+
+// MigrationMap defines a mapping from a version to a transformation plan.
+type MigrationMap map[string]transform.Plan
+
+var Migrations = MigrationMap{
+	"v0.45": nil,
```

Comment:
So, here I wonder what is best, as the tool works standalone as well.
I think there are a few options:

1. store the config files directly in the tool
    - pro: can work standalone
    - con: need to add the file in the tool at each release, as well for minor release with configuration changes
2. use the SDK version of the app and marshal to config, and have a diff
   - pro: we don't mind about the different versions, as we just diff and add what is missing
   - con: as we use the SDK version probably work with only one version, especially standalone
3. write migrations per versions
   - pro: can work standalone, similar to genesis migrate
   - con: need to detect which version you are migrating from (or assuming you can only bump one by one version - like genesis migrate), need to be updated at every configuration change

I like 1 and 3 the best.

Reply:
Step one is already implemented. Working on supporting 2 (as per https://github.com/cosmos/cosmos-sdk/pull/14342#discussion_r1063483118).